pkg/util/mount/mount.go

```diff
@@ -295,22 +295,36 @@ func isBind(options []string) (bool, []string) {
 	// volume mount to be read only.
 	bindRemountOpts := []string{"bind", "remount"}
 	bind := false
+	bindOpts := []string{"bind"}
 
-	if len(options) != 0 {
-		for _, option := range options {
-			switch option {
-			case "bind":
-				bind = true
-				break
-			case "remount":
-				break
-			default:
-				bindRemountOpts = append(bindRemountOpts, option)
-			}
+	// _netdev is a userspace mount option and does not automatically get added when
+	// bind mount is created and hence we must carry it over.
+	if checkForNetDev(options) {
+		bindOpts = append(bindOpts, "_netdev")
+	}
+
+	for _, option := range options {
+		switch option {
+		case "bind":
+			bind = true
+			break
+		case "remount":
+			break
+		default:
+			bindRemountOpts = append(bindRemountOpts, option)
 		}
 	}
 
-	return bind, bindRemountOpts
+	return bind, bindOpts, bindRemountOpts
+}
+
+func checkForNetDev(options []string) bool {
+	for _, option := range options {
+		if option == "_netdev" {
+			return true
+		}
+	}
```

Comment:
NFS has a similar issue, where we can't unmount if share (server is disabled or network issues). Can we apply this to NFS as well?

Reply:
may be. But systemd at least already knows that NFS mounts are using network.  This problem  is more severe in case of block storage volume types, which appear as regular block device to systemd  but actually are networked and hence must be unmounted before network is toredown during shutdown sequence. 

The premise of the PR isn't for Kubernetes to apply `_netdev` whenever it can, but it is to honor that mount option, if user or plugin author has chosen to use it while creating bind mounts.